test/test_torch.py

```diff
@@ -13817,7 +13831,7 @@ def test_ternary_op_mem_overlap(self, device, dtype):
             ("addcdiv", True, True, 'cpu'),
             ("addcdiv", True, True, 'cuda'),
             ("lerp", True, True, 'cpu'),
-            ("lerp", False, False, 'cuda')
+            ("lerp", True, True, 'cuda')
```

Comment:
I did an audit of `TensorIterator::unary_op`, `TensorIterator::binary_op`, and `TensorIterator::nullary_op` cases in the code that use the default. The default before this PR was check_mem_overlap=False, but after this PR, those are True. I found a few that we didn't test, so it would be nice to add tests for them (to check that we've added the checks for both CPU and CUDA:

For: TensorIterator::unary_op
- elu: test in-place variant: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/Activation.cpp#L111-L117
- hardswish: test in-place variant: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/Activation.cpp#L150-L154 
- leaky_relu: test in-place variant https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/Activation.cpp#L713-L735

For TensorIterator::binary_op:
- index_add_: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L375 has two paths.
- atan2: test out variant and in-place variant: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/BinaryOps.cpp#L337-L350 
- ilshift: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/BinaryOps.cpp#L596-L607 
- irshift: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/BinaryOps.cpp#L624-L635
- hypot: test out variant and in-place variant: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/BinaryOps.cpp#L888-L903
- nextafter: test out variant and in-place variant: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/BinaryOps.cpp#L905-L920
- threshold: test in-place variant: https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/Activation.cpp#L380-L392

Reply:
In retrospect, because there are so many ops, I think this PR would be easier to review if we split it up into, e.g.:
- a PR to remove TensorIterator::unary_op's check_mem_overlap (and instead default to True)
- a PR to remove TensorIterator::binary_op's check_mem_overlap (and instead default to True)
- ...
- a PR that hits the switch for the default of `check_mem_overlap` in TensorIterator to True.

However, I'm fine with reviewing all the changes in one single PR (this PR) because the ultimate line count will probably still be manageable.